Vagrantfile

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2021 Red Hat, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This is a Vagrant configuration file.
+#
+
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# Read the Kata Containers CI job from the CI_JOB environment variable.
+job = ENV['CI_JOB'] || ""
+guest_home_dir = '/home/vagrant'
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+
+  config.trigger.before [:up, :ssh] do |trigger|
+    if ENV['GOPATH'] == nil
+      trigger.warn = "You must export the GOPATH environment variable"
+      trigger.abort = true
+    end
+  end
+
+  # By default vagrant sync the current directory. Let's disabled it because the directory
+  # will be synced later to the proper destination.
+  config.vm.synced_folder ".", "/vagrant", disabled:true
+  config.vm.synced_folder ".", "#{guest_home_dir}/go/src/github.com/kata-containers/tests", type:"rsync"
+  config.vm.synced_folder "#{ENV['GOPATH']}/src/github.com/kata-containers/kata-containers",
+    "#{guest_home_dir}/go/src/github.com/kata-containers/kata-containers", type:"rsync"
+
+  config.vm.provider "libvirt" do |lv|
+    lv.driver = "kvm"
+    lv.cpus = "4"
+    lv.memory = "8192"
+  end
+
+  # Shared provision script.
+  config.vm.provision "shell", env: {"CI_JOB" => job}, inline: <<-SHELL
+
+    export GOPATH="#{guest_home_dir}/go"
+    kata_tests_repo_dir="${GOPATH}/src/github.com/kata-containers/tests"
+
+    env_file="#{guest_home_dir}/ci_job_env"
+    cat <<-EOF > ${env_file}
+export GOPATH="$GOPATH"
+export PATH="/usr/local/go/bin:\$GOPATH/bin:\$PATH"
+export CI="true"
+export CI_JOB="$CI_JOB"
+pushd $kata_tests_repo_dir &>/dev/null
+source .ci/ci_job_flags.sh
+popd &>/dev/null
+EOF
+
+    source ${env_file}
+
+    # Customize the .bashrc so that it will have the variables exported
+    # after log-in.
+    echo "cd $kata_tests_repo_dir" >> #{guest_home_dir}/.bashrc
+    echo "source $env_file" >> #{guest_home_dir}/.bashrc
+
+    distro=$(source /etc/os-release; echo $ID)
+    case "$distro" in
+      "fedora")
+        # Fedora >= 32 Kernel comes configured with cgroup v2 by default.
+        # This switches back to cgroup v1. It requires a reboot.
+        sudo dnf install -y grubby
+        sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+        ;;
+      "ubuntu")
+        # TODO: redis-server package fails to install if IPv6 is disabled. Move this
+        # code to the setup script.
+        if [[ $(sudo sysctl -n net.ipv6.conf.all.disable_ipv6) -eq 1 ]]; then
+          sudo sysctl -w net.ipv6.conf.all.disable_ipv6=0
+          sudo sed -i 's/\(net.ipv6.conf.all.disable_ipv6\).*/\1=0/' /etc/sysctl.conf
```

Comment:
Does redis add this entry to `sysctl.conf`? If not, this won't work as it requires the value to already be specified in this file (but it might have been specified in the grub config for example). I guess it "works" in the sense that you've manually disabled ipv6 in the command above, but still potentially confusing.

Reply:
@jodh-intel I am not sure I understood your comment. apt-get refuses to install the redis-server package if IPv6 is disable; the command above ensure IPv6 is *enabled*, therefore the package gets installed. 

It seems redis is only needed by the `tests/integration/kubernetes/k8s-port-forward.bats` test which, btw, is currently being skipped.... let me run it to see what happens.

Reply:
There's no way to avoid forcing IPv6? if not, the host probably should also requires it, i'd mentioned it in the README